lib/ably/rest/channel.rb

```diff
@@ -164,14 +163,21 @@ def presence
         @presence ||= Presence.new(client, self)
       end
 
+      # Sets or updates the stored channel options. (#RSL7)
+      # @param channel_options [Hash, Ably::Models::ChannelOptions]  A hash of options or a {Ably::Models::ChannelOptions}
+      # @return [Ably::Models::ChannelOptions]
+      def set_options(channel_options)
+        update_options(channel_options)
+      end
+      alias options= set_options
+
+      private
+
       # @api private
       def update_options(channel_options)
```

Comment:
@lukaszsliwa I thought of making it private since we now have `set_options` and `options=` which are defined by the specification and making a third alias that isn't in spec I think is incorrect

Reply:
```
(RSL7) Channel#setOptions takes a ChannelOptions object and sets or updates the stored channel options, then indicates success
```

I recommend something like this as it says to take `ChannelOptions` object. Now, it takes another method `update_options(...)` and it looks strange. There is no profit of using private method here.

```ruby
# Sets or updates the stored channel options. (#RSL7)
# @param channel_options [Hash, Ably::Models::ChannelOptions]  A hash of options or a {Ably::Models::ChannelOptions}
# @return [Ably::Models::ChannelOptions]
def set_options(channel_options)
  @options = Ably::Models::ChannelOptions(channel_options) # "takes a ChannelOptions object"
end
alias options= set_options
alias update_options set_options
```

`alias update_options set_options` will help with backward compatibility


In `lib/ably/realtime/channel.rb` is similar case.